src/org/exist/Indexer.java

```diff
@@ -305,6 +305,9 @@ public void endDTD() {
     @Override
     public void endDocument() {
         if (!validate) {
+            if(indexListener != null) {
+                indexListener.endDocument(transaction);
```

Comment:
@shabanovd You mean inside `appendChildren`? If so, yes I could see that might be confusing, but then I might move such calls around a block of XQuery Update to a document.